daemon/create.go

```diff
@@ -152,5 +153,10 @@ func (daemon *Daemon) VolumeCreate(ctx context.Context, name, driverName string,
 	if err != nil {
 		return nil, err
 	}
+
+	// keep "docker run -v existing_volume:/foo --volume-driver other_driver" work
+	if (driverName != "" && v.DriverName() != driverName) || (driverName == "" && v.DriverName() != volume.DefaultDriverName) {
```

Comment:
IMHO, I think it just did the same thing as "volumes.Create" does. 
func (s *VolumeStore) Create(name, driverName string, opts map[string]string) (volume.Volume, error) {
	s.mu.Lock()
	if vc, exists := s.vols[name]; exists {
		v := vc.Volume
		s.mu.Unlock()
		return v, nil
	}
Create() just do the same check and return the existed volume if it exists. Is it necessary to do the same thing?